### DISCUSSION TITLE
Exemple of relation property

### DISCUSSION THREAD

**Discussion by iizno** [mariosimao/notion-sdk-php#381] [Q&A] [answered]
Hi ! Thanks for this package, it's really usefull to us :) 

I'm almost done with anything I needed but i'm struggling with the relation property. The property already exist, and is working on Notion in the Database, but I can't figure out how to set the page ID in the field .. 

If anyone has an idea 🤷 ?

"$this->parent->notion_page_id" contains the page_id of the database row I want to link. 

![image](https://github.com/user-attachments/assets/f61eb16b-2b01-4ab9-895b-897a9a5de1f5)

**Comment by mariosimao** [comment]
Hi! I think this should work.

```php
$page->addProperty('merge_id', Relation::create($this->parent->notion_page_id));
```

**Reply by iizno** [reply]
Sadly, not. I tried with Notion\Pages\Properties\Relation
But it doesn't work (no error, but the field in Notion stays empty).

I debugued the $page variable and I can see my merge_id here, but the pageIds seems empty : 
![image](https://github.com/user-attachments/assets/b5100584-ab8c-4e10-930c-4d2573eed0c1)

my notion_page_id are not empty, I checked : 
![image](https://github.com/user-attachments/assets/a515dfb8-a17b-48c9-b0f7-7841f6a3568b)

I tried with  \Notion\Databases\Properties\Relation but it doesn't seems related.

**Reply by iizno** [reply]
So I investigated a little further : 
![image](https://github.com/user-attachments/assets/e607b948-44cc-42b3-a7e2-b97e58cc23a4)

The first dump get me this : 
![image](https://github.com/user-attachments/assets/ac396317-6255-49ca-9fcf-3b8c39cb652e)

but the second one get me this : 
![image](https://github.com/user-attachments/assets/122d22c3-2dd8-48eb-95c9-1bcf6174f1fd)

**Comment by iizno** [comment, marked as answer]
Ok, I found the issue. 

It's                 

```php
$page = $page->addProperty('merge_id', Relation::create($this->parent->notion_page_id));
```

and not 

```php
$page->addProperty('merge_id', Relation::create($this->parent->notion_page_id));
```
Sorry for wasting your time, and again, thanks for this awesome package.
